### ISSUE TITLE
WordPress Gutenberg Block Library CSS incorrect validation

### ISSUE THREAD

**Issue by umangbhanvadia-multidots** [validator/validator#1698]
URL being validated or code to reproduce error:
```css
h1.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h1.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]),h2.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h2.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]),h3.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h3.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]),h4.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h4.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]),h5.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h5.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]),h6.has-text-align-left[style*=writing-mode]:where([style*=vertical-lr]),h6.has-text-align-right[style*=writing-mode]:where([style*=vertical-rl]){rotate:180deg}

.wp-lightbox-overlay .close-button{align-items:center;cursor:pointer;display:flex;justify-content:center;min-height:40px;min-width:40px;padding:0;position:absolute;right:calc(env(safe-area-inset-right) + 16px);top:calc(env(safe-area-inset-top) + 16px);z-index:5000000}

.wp-block-navigation__responsive-container.is-menu-open{animation:overlay-menu__fade-in-animation .1s ease-out;animation-fill-mode:forwards;background-color:inherit;display:flex;flex-direction:column;overflow:auto;padding:clamp(1rem,var(--wp--style--root--padding-top),20rem) clamp(1rem,var(--wp--style--root--padding-right),20rem) clamp(1rem,var(--wp--style--root--padding-bottom),20rem) clamp(1rem,var(--wp--style--root--padding-left),20em);z-index:100000}

p.has-text-align-left[style*="writing-mode:vertical-lr"],p.has-text-align-right[style*="writing-mode:vertical-rl"]{rotate:180deg}`
```
paste error messages into code block
![CleanShot 2024-03-01 at 16 23 15@2x](https://github.com/validator/validator/assets/53076948/c0b5aed0-4dc1-4e40-85f1-561fbca2edf7)

**Comment by nschonni** [comment]
You don't say what the problem is, but I tried to look at it because I was currious. I think there are bad uses of `:where`. I tried using prettier just to make it more readable, but it also can't parse this CSS. Might be something in the minification process that's removing spaces or something else that is making this invalid

**Comment by umangbhanvadia-multidots** [comment]
Hello @nschonni,

Sorry for the inconvenience. Actually I have attached the CSS which are exists in the core file of WordPress framework and while we check the site in the NU HTML Checker tool, We face these errors. I am mentioning the CSS related errors which we are facing. 

**Errors:**

- Error: CSS: rotate: Property rotate doesn't exist.
- Error: CSS: right: Parse Error.
- Error: CSS: top: Parse Error.
- Error: CSS: padding: The types are incompatible.

Please share your feedback regarding this.

**Comment by ylafon** [comment]
The padding issue, related to variables handling in clamp() has been fixed upstream. See https://github.com/w3c/css-validator/issues/412 
`rotate`, CSS Transform Module 2 in general, is not implemented yet.

**Comment by j9t** [comment]
I can still [reproduce](https://validator.w3.org/nu/#textarea) some parse error related to the `env()` function. This may be due to this still being [in draft status](https://drafts.csswg.org/css-env/#env-function)—@ylafon, would this be accurate, or is there something else to consider here? (Thanks!)

**Comment by sideshowbarker** [comment]
Please re-raise this issue against the CSS validator at https://github.com/w3c/css-validator/issues/new.

(For CSS checking, the HTML checker relies uses the CSS validator code as a library — so after any change for this gets made to the upstream CSS-validator code, we will just pull that change into the HTML checker.)
